keanu-project/src/test/java/io/improbable/keanu/network/LambdaSectionTraversalTest.java

```diff
@@ -70,4 +71,24 @@ public void doesGetAllUpstreamVertices() {
         assertEquals(5, verticesDepthFirst.size());
         assertThat(verticesDepthFirst, containsInAnyOrder(A, B, aPlusB, cSigma, C));
     }
+
+    @Test
+    public void doesGetUpstreamProbabilisticVerticesOfCollection() {
+
+        LambdaSection lambdaSection = LambdaSection.getUpstreamLambdaSectionForCollection(Arrays.asList(C, cSigma), false);
+        Set<Vertex> verticesDepthFirst = lambdaSection.getLatentAndObservedVertices();
+
+        assertEquals(3, verticesDepthFirst.size());
```

Comment:
you don't need to check the size, as that's done implicitly in `containsInAnyOrder` on the next line

Reply:
... and then I'd suggest you inline `verticesDepthFirst` too, and your test is only 2 lines:

    @Test
    public void doesGetUpstreamProbabilisticVerticesOfCollection() {
        LambdaSection lambdaSection = LambdaSection.getUpstreamLambdaSectionForCollection(Arrays.asList(C, cSigma), false);
        assertThat(lambdaSection.getLatentAndObservedVertices(), containsInAnyOrder(A, B, C));
    }

Reply:
ditto for every test in this file